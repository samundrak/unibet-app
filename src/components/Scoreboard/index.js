import React from 'react';
import {
  ScoreItem,
  Container,
  Score,
  TeamLabelBox,
  TeamName,
  Time,
} from './style';

class ScoreBoard extends React.Component {
  render() {
    return (
      <Container>
        <Score>
          <ScoreItem>15</ScoreItem>-<ScoreItem>{this.props.index}</ScoreItem>
        </Score>
        <TeamLabelBox>
          <TeamName>Bedene, Aljaz</TeamName>-<TeamName> Daniel, Taro</TeamName>
        </TeamLabelBox>
        <Time>Today, 14:12</Time>
      </Container>
    );
  }
}
export default ScoreBoard;
