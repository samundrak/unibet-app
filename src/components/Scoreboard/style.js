import styled from 'styled-components';

const MARGIN_BETWEEN_TOP_ITEMS = '2%';
const Container = styled.div`
  width: inherit;
  height: inherit;
  background-color: #222222;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
`;
const Score = styled.div`
  margin-top: ${MARGIN_BETWEEN_TOP_ITEMS};
  width: inherit;
  justify-content: center;
  align-items: center;
  display: flex;
  color: #e6ab38;
  font-size: 2em;
`;
const ScoreItem = styled.div`
  font-weight: bold;
  margin-right: 1%;
  margin-left: 1%;
`;
const TeamName = styled.div`
  margin-right: 1%;
  margin-left: 1%;
`;
const TeamLabelBox = styled.div`
  color: white;
  margin-top: ${MARGIN_BETWEEN_TOP_ITEMS};
  width: inherit;
  justify-content: center;
  align-items: center;
  display: flex;
  font-weight: bold;
  font-size: 1.5em;
`;
const Time = styled.div`
  margin-top: ${MARGIN_BETWEEN_TOP_ITEMS};
  font-weight: bold;
  color: #867b61;
  font-size: 1em;
`;
const Gap = styled.div`
  margin-top: ${MARGIN_BETWEEN_TOP_ITEMS};
`;
export { Gap, TeamName, ScoreItem, Container, Score, TeamLabelBox, Time };
