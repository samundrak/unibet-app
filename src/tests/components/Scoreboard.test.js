import React from 'react';
import { shallow } from 'enzyme';
import Scoreboard from '.././../components/Scoreboard';

const data = {
  homeName: 'samundra',
  awayName: 'notsamundra',
  sport: 'football',
  start: new Date().toString(),
  score: {
    home: '1',
    away: '1',
  },
};
describe('test scoreboard', () => {
  it('should render scorboard', () => {
    const wrapper = shallow(<Scoreboard event={data} />);
    expect(wrapper.html()).toMatchSnapshot();
  });
});
