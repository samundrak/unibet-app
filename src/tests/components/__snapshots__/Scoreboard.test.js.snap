// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test scoreboard should render scorboard 1`] = `"<div class=\\"sc-bdVaJa gvahtW\\"><div class=\\"sc-bwzfXH duNtlP\\"><div class=\\"sc-htpNat kUJhXk\\">1</div>-<div class=\\"sc-htpNat kUJhXk\\">1</div></div><div class=\\"sc-ifAKCX bNEQGx\\"><img src=\\"/icons/football.png\\" alt=\\"football\\"/><div class=\\"sc-bxivhb fnFrdH\\">samundra</div>-<div class=\\"sc-bxivhb fnFrdH\\"> notsamundra</div></div><div class=\\"sc-EHOje gHoUDM\\">2019-04-26</div><div class=\\"sc-bZQynM kSOmQx\\"><button class=\\"sc-gzVnrw kcIzgI\\">Place a Bet</button></div></div>"`;
